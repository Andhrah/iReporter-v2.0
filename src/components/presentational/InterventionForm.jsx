import React from 'react';
import PropTypes from 'prop-types';

const InterventionForm = ({
  handleChange, handleSubmit, values,
}) => {
  const {
    economy,
    badroad,
    badHousing,
    noElectricity,
    poorElectricity,
    refuseDamping,
    collapsedBridge,
    otherReasons,
    otherReasonsSpecify,
    interventionDate,
    describe,
    interventionImages,
  } = values;

  return (
    <div className="report-form-container">
      <h3>
      Call for Government
        {' '}
        <span>Intervention</span>
      </h3>
      <form onSubmit={handleSubmit} className="report-form">
        <p>Reasons for intervention</p>
        <div className="first-checkbox-container">
          <div className="checkbox-container">
            <input type="checkbox" value={economy} name="economy" onChange={handleChange} />
            <label htmlFor="economy">Economy</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={badroad} name="badroad" onChange={handleChange} />
            <label htmlFor="badroad">Bad road</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={noElectricity} name="noElectricity" onChange={handleChange} />
            <label htmlFor="noElectricity">No electricity</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={poorElectricity} name="poorElectricity" onChange={handleChange} />
            <label htmlFor="poorElectricity">Poor electricity</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={refuseDamping} name="refuseDamping" onChange={handleChange} />
            <label htmlFor="refuseDamping">No provision refuse damping</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={badHousing} name="badHousing" onChange={handleChange} />
            <label htmlFor="badHousing">Housing under power tension</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={collapsedBridge} name="collapsedBridge" onChange={handleChange} />
            <label htmlFor="collapsedBridge">Collapse</label>
          </div>
          <div className="checkbox-container">
            <input type="checkbox" value={otherReasons} name="otherReasons" onChange={handleChange} />
            <label htmlFor="otherReasons">Others</label>
          </div>
        </div>
        <div className="checkbox-container">
          <label htmlFor="otherReasonsSpecify">If others, please specify</label>
          <input type="text" value={otherReasonsSpecify} name="otherReasonsSpecify" onChange={handleChange} />
        </div>
        {/* ----------------------------------------------------------------------- */}
        {/* where it happened */}
        {/* --------------------------------------------------------------------- */}
        <p>Where did it happen?</p>
        <label htmlFor="namesInvolved">Location</label>
        <input style={{ height: 40, paddingLeft: 15, paddingRight: 15, fontSize: 17 }} name="displayLocation" onChange={handleChange} id="displayLocation" required />

        {/* <div className="map">Google Map</div> */}
        {/* ----------------------------------------------------- */}
        {/* when did it happen */}
        {/* ----------------------------------------------------- */}
        {/* <div className="checkbox-container">
          <label htmlFor="interventionDate">When did it happen?</label>
          <input type="text" value={interventionDate} name="interventionDate" onChange={handleChange} />
        </div> */}
        {/* --------------------------------------------------------------*/}
        {/* what happened */}
        {/* -------------------------------------------------------------*/}
        <div className="checkbox-container">
          <p>Please describe what happened?</p>
          <textarea name="describe" value={describe} onChange={handleChange} cols={50} rows={10} placeholder="Enter text here" />
        </div>
        {/* ----------------------------------------------------- */}
        {/* Add image(s) or Video */}
        {/* ----------------------------------------------------- */}
        {/* <div className="checkbox-container"> */}
        {/* <p>Add image or video related to reason for intervention</p> */}
        {/* <input value={interventionImages} name="interventionImages" onChange={handleChange} type="file" className="custom-file-input" /> */}
        {/* </div> */}
        <button type="submit" className="report-button">Call For Intervention</button>
      </form>
    </div>
  );
};

InterventionForm.propTypes = {
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  values: PropTypes.object,
  // errors: PropTypes.object,
};

InterventionForm.defaultProps = {
  handleChange: () => {},
  handleSubmit: () => {},
  values: {},
  // errors: {},
};

export default InterventionForm;
