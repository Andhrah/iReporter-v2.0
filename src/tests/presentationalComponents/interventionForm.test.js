import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import InterventionForm from '../../components/presentational/InterventionForm';

const props = {
  handleChange: () => {},
  handleSubmit: () => {},
  values: {
    economy: 'economy',
    badroad: 'badroad',
    badHousing: 'badHousing',
    noElectricity: 'noElectricity',
    poorElectricity: 'poorElectricity',
    refuseDamping: 'refuseDamping',
    collapsedBridge: 'collapsedBridge',
    otherReasons: 'nothing',
    otherReasonsSpecify: 'nothing',
    interventionDate: '2019-05=04',
    describe: 'describe',
    interventionImages: 'interventionImages.png',
  },
};

describe('InterventionForm  test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <InterventionForm {...props} />
      </MemoryRouter>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

});
