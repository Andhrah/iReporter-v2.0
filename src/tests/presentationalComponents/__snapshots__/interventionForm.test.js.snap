// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterventionForm  test suite renders correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "skf8lw",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "skf8lw",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <InterventionForm
      handleChange={[Function]}
      handleSubmit={[Function]}
      values={
        Object {
          "badHousing": "badHousing",
          "badroad": "badroad",
          "collapsedBridge": "collapsedBridge",
          "describe": "describe",
          "economy": "economy",
          "interventionDate": "2019-05=04",
          "interventionImages": "interventionImages.png",
          "noElectricity": "noElectricity",
          "otherReasons": "nothing",
          "otherReasonsSpecify": "nothing",
          "poorElectricity": "poorElectricity",
          "refuseDamping": "refuseDamping",
        }
      }
    >
      <div
        className="report-form-container"
      >
        <h3>
          Call for Government
           
          <span>
            Intervention
          </span>
        </h3>
        <form
          className="report-form"
          onSubmit={[Function]}
        >
          <p>
            Reasons for intervention
          </p>
          <div
            className="first-checkbox-container"
          >
            <div
              className="checkbox-container"
            >
              <input
                name="economy"
                onChange={[Function]}
                type="checkbox"
                value="economy"
              />
              <label
                htmlFor="economy"
              >
                Economy
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="badroad"
                onChange={[Function]}
                type="checkbox"
                value="badroad"
              />
              <label
                htmlFor="badroad"
              >
                Bad road
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="noElectricity"
                onChange={[Function]}
                type="checkbox"
                value="noElectricity"
              />
              <label
                htmlFor="noElectricity"
              >
                No electricity
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="poorElectricity"
                onChange={[Function]}
                type="checkbox"
                value="poorElectricity"
              />
              <label
                htmlFor="poorElectricity"
              >
                Poor electricity
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="refuseDamping"
                onChange={[Function]}
                type="checkbox"
                value="refuseDamping"
              />
              <label
                htmlFor="refuseDamping"
              >
                No provision refuse damping
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="badHousing"
                onChange={[Function]}
                type="checkbox"
                value="badHousing"
              />
              <label
                htmlFor="badHousing"
              >
                Housing under power tension
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="collapsedBridge"
                onChange={[Function]}
                type="checkbox"
                value="collapsedBridge"
              />
              <label
                htmlFor="collapsedBridge"
              >
                Collapse
              </label>
            </div>
            <div
              className="checkbox-container"
            >
              <input
                name="otherReasons"
                onChange={[Function]}
                type="checkbox"
                value="nothing"
              />
              <label
                htmlFor="otherReasons"
              >
                Others
              </label>
            </div>
          </div>
          <div
            className="checkbox-container"
          >
            <label
              htmlFor="otherReasonsSpecify"
            >
              If others, please specify
            </label>
            <input
              name="otherReasonsSpecify"
              onChange={[Function]}
              type="text"
              value="nothing"
            />
          </div>
          <p>
            Where did it happen?
          </p>
          <label
            htmlFor="namesInvolved"
          >
            Location
          </label>
          <input
            id="names-involved"
            name="namesInvolved"
            onChange={[Function]}
            required={true}
            style={
              Object {
                "fontSize": 17,
                "height": 40,
                "paddingLeft": 15,
                "paddingRight": 15,
              }
            }
          />
          <div
            className="checkbox-container"
          >
            <p>
              Please describe what happened?
            </p>
            <textarea
              cols={50}
              name="describe"
              onChange={[Function]}
              placeholder="Enter text here"
              rows={10}
              value="describe"
            />
          </div>
          <button
            className="report-button"
            type="submit"
          >
            Call For Intervention
          </button>
        </form>
      </div>
    </InterventionForm>
  </Router>
</MemoryRouter>
`;
